src/createRollupFetchTransactionHash.ts

```diff
@@ -0,0 +1,47 @@
+import { deterministicFactoriesDeploymentEstimatedFees } from './constants';
+import { Address, PublicClient } from 'viem';
+import { AbiEvent } from 'abitype'
+
+const RollupInitializedEventAbi: AbiEvent = {
+  anonymous: false,
+  inputs: [
+      {
+          indexed: false,
+          internalType: "bytes32",
+          name: "machineHash",
+          type: "bytes32",
+      },
+      {
+          indexed: false,
+          internalType: "uint256",
+          name: "chainId",
+          type: "uint256",
+      },
+  ],
+  name: "RollupInitialized",
+  type: "event",
+};
+
+export async function createRollupFetchTransactionHash(
+  rollupAddress: Address,
+  publicClient: PublicClient,
+) {
+  // Find the RollupInitialized event from that Rollup contract
+  const rollupInitializedEvents = await publicClient.getLogs({
+    address: rollupAddress,
+    event: RollupInitializedEventAbi,
+    fromBlock: 'earliest',
```

Comment:
Instead of fetching from `earliest`, maybe we can add the block number for when the rollup creator was deployed at to `wagmi.config.ts` and fetch from then?

Reply:
Makes sense, but I think I'd have to modify the object then, because each contract in each chain has been deployed at a different block height. Should we create a different PR for that?